GVFS/GVFS.Platform.Mac/MacServiceProcess.cs

```diff
@@ -0,0 +1,77 @@
+using GVFS.Common;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Platform.Mac
+{
+    /// <summary>
+    /// Class to query the configured services on macOS
+    /// </summary>
+    public class MacServiceProcess
+    {
+        private const string LaunchCtlPath = @"/bin/launchctl";
+        private const string LaunchCtlArg = @"list";
+
+        private IProcessHelper processHelper;
+
+        public MacServiceProcess()
+        : this(new ProcessHelperImpl())
+        {
+        }
+
+        public MacServiceProcess(IProcessHelper processHelper)
+        {
+            this.processHelper = processHelper;
+        }
+
+        public bool TryGetServices(out List<ServiceInfo> services, out string error)
+        {
+            ProcessResult result = this.processHelper.Run(LaunchCtlPath, LaunchCtlArg, true);

```

Comment:
Does this return expected results when `gvfs upgrade --confirm` is run elevated? `GVFS.Service` on the Mac is run per user, so running as root might return a list with no `GVFS.Service` in it (unless root user has logged-in as well through login window.)

Reply:
hmm... that might be the case... I will investigate this more - do you have suggestions for an alternative here?